Use the SDK and better jinja2

jhkennedy opened the pull request.

jhkennedy committed:
Rework jinja templates and autorift test for SDK

jhkennedy committed:
Update InSAR, RTC to use SDK

jhkennedy committed:
fix lint

asjohnston-asf commented on hyp3_testing/helpers.py:
```diff
@@ -1,75 +1,13 @@
-        if is_zipfile(product_file):
-            with ZipFile(product_file) as zip_:
-                zip_.extractall(path=directory)
+def extract_products(products: List[Path]):
```
`product_files` or `product_zip_files` might be a better name

asjohnston-asf commented on hyp3_testing/util.py:
```diff
@@ -0,0 +1,29 @@
+from jinja2 import Environment, PackageLoader, StrictUndefined, select_autoescape
+
+
+def get_job_name() -> str:
```
`generate` instead of `get`? I initial expected this function to decode something

asjohnston-asf commented on hyp3_testing/util.py:
```diff
@@ -0,0 +1,29 @@
+    return env
+
+
+def render_template(template: str, **kwargs) -> dict:
```
`template_file` might be better than `template`, I wasn't sure if it was the file or the content at first glance

asjohnston-asf commented on setup.py:
```diff
@@ -32,11 +32,10 @@
 
     install_requires=[
-        'hyp3lib',
+        'hyp3_sdk',
```
```suggestion
        'hyp3_sdk>=1.1',
```

asjohnston-asf commented on tests/conftest.py:
```diff
@@ -30,16 +32,26 @@ def comparison_dirs(tmp_path_factory, golden_dirs):
             path = Path(dir_)
-            if not path.exists():
-                path.mkdir()
+            assert path.name in ('main', 'develop')
```
What's the motivation for limiting the dir names here? I've been in the habit of naming my dirs `prod` and `test`

asjohnston-asf replied:
Hmm, I see it's so the `zip()` calls later always correctly pair the right HyP3 session with the corresponding directories. I'd prefer if things weren't coupled that tightly, but I'm yet to think of a better approach.

jhkennedy replied:
refactored to assume comparison dirs are [golden, test] as is already documented in our readme so I don't need this assert, and bundled hyp3 APIs with this fixture (now `comparison_environments`)

asjohnston-asf commented on tests/test_tifs.py:
```diff
@@ -51,46 +49,41 @@ def _get_tif_tolerances(file_name: str):
+        jobs = hyp3.find_jobs(name=job_name)
+        jobs = hyp3.watch(jobs)
+        products = jobs.download_files(dir_)
+        helpers.extract_products(products)
```
Should consider deleting the `.zip` files after they're extracted, but I assume we'll leave that for the next PR since it will mean refactoring how we generate our list of products to compare to look at directories instead of zips.

jhkennedy replied:
:+1: I'm going to wait until we work on the disk usage as that'll be coupled with it (could make it so only InSAR would need it at that point)

asjohnston-asf commented on tests/test_util.py:
```diff
@@ -0,0 +1,18 @@
+    with pytest.raises(UndefinedError):
+        util.render_template('insar_gamma_golden.json.j2')
+
+    util.render_template('insar_gamma_golden.json.j2', name='test')
```
does the new `helpers.extract_products()` function need a test as well?

jhkennedy replied:
/shug

jhkennedy committed:
combine comparison dirs and hyp3s

jhkennedy committed:
better names

jhkennedy committed:
specify hyp3_sdk requirements

Co-authored-by: Andrew Johnston <asjohnston@alaska.edu>

jhkennedy committed:
Install netCDF package so xarray can open netCDF files

jhkennedy committed:
whitespace

jhkennedy committed:
list zip to resue the iterator

jhkennedy commented:
~~Golden Tests running for this branch:~~
* ~~[autoRIFT](https://github.com/ASFHyP3/hyp3-testing/actions/runs/961758026)~~
* ~~[RTC](https://github.com/ASFHyP3/hyp3-testing/actions/runs/961759557)~~
* ~~[InSAR](https://github.com/ASFHyP3/hyp3-testing/actions/runs/961766939)~~

All failed with

```
Requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='hyp3-test-api.asf.alaska.edu', port=443): Read timed out. (read timeout=None)
```
after waiting on prod jobs to finish, *then* trying to talk to test

jhkennedy committed:
Re-connect to HyP3 each time

`*_wait_and_download` test were failing with a requests.exceptions.ReadTimeout
due to the long period of inactivity between authenticating to
hyp3-test, waiting for hyp3-prod jobs to finish, and then searching for
hyp3-test jobs.

Instead of trying to keep the hyp3-test instance open, just reconnect
each time we're going to use it.

jhkennedy commented:
Golden Tests running for this branch:
* [RTC](https://github.com/ASFHyP3/hyp3-testing/actions/runs/962238977) -- PASSED
* [InSAR](https://github.com/ASFHyP3/hyp3-testing/actions/runs/962412185) -- OK (failed in an expected way)
* [autoRIFT](https://github.com/ASFHyP3/hyp3-testing/actions/runs/962410710) -- OK (failed in an expected way)

Jlrine2 commented on tests/conftest.py:
```diff
@@ -1,6 +1,7 @@
 import shutil
 from pathlib import Path
 
+import hyp3_sdk as sdk
```
I don't feel very strongly about it, but I prefer not to use `as` imports unless it is common convention,

jhkennedy replied:
I do this myself out of habit b/c I find the number of "hyp3"s in common commands super annoying:

```python
hyp3 = hyp3_sdk.HyP3(hyp3_sdk.HYP3_TEST)
```

I think this reads much better:

```python
hyp3 = sdk.HyP3(sdk.HYP3_TEST)
```
but in general, I agree as it's pretty brazen of us to steal the generic `sdk` namespace, so done.

jhkennedy committed:
sdk less than 2 to gaurd against major package upgrades

jhkennedy committed:
no as sdk
